crates/sui-data-ingestion-core/src/remote_fetcher/hybrid_fetcher.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::remote_fetcher::object_store_fetcher::ObjectStoreFetcher;
+use crate::remote_fetcher::rest_fetcher::RestFetcher;
+use crate::remote_fetcher::RemoteFetcherTrait;
+use std::sync::Arc;
+use sui_types::full_checkpoint_content::CheckpointData;
+use sui_types::messages_checkpoint::CheckpointSequenceNumber;
+
+/// The HybridFetcher first tries to fetch the checkpoint from the REST API.
+/// If that fails, it falls back to fetching the checkpoint from the archival store.
+pub(crate) struct HybridFetcher {
+    archival_fetcher: ObjectStoreFetcher,
+    rest_fetcher: RestFetcher,
+}
```

Comment:
Can we just have this be generic with a vec of fetcher trait objects? then you can just add any fetcher and not just these two (for the future)